### ISSUE TITLE
Consistently update schedule A and B data

### ISSUE THREAD

**Issue by LindsayYoung** [18F/openFEC#1820]
Updates failing because it is trying to add columns that already exist .

I think we just want to get rid of the things in the old que before adding the things in the new que, that way, replacements should run smoothly. 
https://github.com/18F/openFEC/blob/develop/data/sql_setup/prepare_schedule_a.sql#L37

Here is the error:
From Dev Sun, Jul 17, 2016 at 5:06 AM
```
2016-07-17 09:06:45 ERROR:manager:(psycopg2.IntegrityError) duplicate key value violates unique constraint "ofec_sched_a_2013_2014_tmp_pkey"
DETAIL:  Key (sub_id)=(2071520161306759407) already exists.
```

**Comment by ccostino** [comment]
At least we're now seeing the primary key constraint violations we were expecting to see!

**Comment by LindsayYoung** [comment]
I don't think the issue is what I thought it was, I am seeing a record identified by sub_id that is in the existing table and new queue table, but not the old table. 

Now I am looking closer at how our queues are made.

**Comment by jontours** [comment]
Well, not that it helps or anything, but we know that must be happing in an insert statement... Just a question of why that insert even happens, or if it should happen.

**Comment by LindsayYoung** [comment]
there are 78200 records in the schedule a new queue and it looks like 78185 of them have already been added

**Comment by ccostino** [comment]
This goes back to the notion that the queue tables aren't being properly cleaned out.  Is it possible that this is the case due to the weekly update?  We fixed this in at least `dev` though by changing it back to `update_all`, which does properly clear out the queues, correct?

Based on the `fec-update-bot` notifications from last night it does look like `dev` is okay now, but staging is still affected.

**Comment by ccostino** [comment]
@LindsayYoung, were those counts in `dev` or `staging`?

**Comment by LindsayYoung** [comment]
Those counts were in dev.

I think the difference is that I ran rebuild_aggregates() on dev yesterday but not staging. The timing suggests that the issue was the weekly rebuild, but I remember we both checked to confirm that that cleared the queue. I can double check that again. 

I'll rebuild the aggregates on staging and see if I can get updates going on staging. Tonight we can try a nightly update on staging and a weekly rebuild on dev and see how it goes.

**Comment by ccostino** [comment]
👍 to all of that, thanks! :-)

The weekly rebuild is likely the issue in the case since that was running before we made the `update_all` change in it, which does clear the queue (`update_all` includes calls to the `update_itemized` method which runs the corresponding scripts in the `sql_setup` folder - those scripts drop and recreate the queue tables).

Hopefully after rebuilding the aggregates on staging we'll see some parity again!

**Comment by jontours** [comment]
I don't know if y'all have talked about this, but is there any reason why prepare_schedule_a.sql is missing these lines that are in prepare_schedule_b.sql? 

https://github.com/18F/openFEC/blob/develop/data/sql_setup/prepare_schedule_b.sql#L41

**Comment by LindsayYoung** [comment]
@jontours that is a great observation! We should definitely figure that out.

**Comment by ccostino** [comment]
D'oh!  It definitely looks like they should be there and that would certainly explain why the schedule A tables aren't being cleared out even on a nightly update, as @LindsayYoung noted above!

I looked through a history of the file and they were removed quite a while back in a refactoring it looks like.  Furthermore, it's not just those two lines - every case is missing the `delete` statements that are present in the `prepare_schedule_b.sql` file.

In addition to this, it also sheds light on where the `timestamp` came from and why it was missing from the schedule B SQL when I made the updates to support the `two_year_transaction_period`.

Big +1 to investigating this and figuring out the correct course of action here.  I suspect we may still need these `delete` statements given the behavior we're seeing?

**Comment by jontours** [comment]
Yeah I noticed @jmcarp took those lines out in December, and if they were taken out then those deletes were either accounted for somewhere else, or were just not needed.  But it is an interesting difference, considering the update process should be the same.  Also since the queue tables have no primary keys, there is nothing to stop from inserting multiple records with the same primary key.  I don't know if it's the root cause but it's worth thinking about.

**Comment by noahmanger** [comment]
Is this done?

**Comment by ccostino** [comment]
No, this isn't done yet.  We haven't had a chance to come back to it yet with all of the production issues we were working through.

**Comment by ccostino** [comment]
It does appear to me that we just need to add these delete statements back into the trigger for the schedule A queue tables.  If I'm not missing anything else about this, we can have this done today.

**Comment by noahmanger** [comment]
Resolved by #1849
